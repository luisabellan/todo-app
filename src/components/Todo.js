import React from "react";

export default function Todo(props) {
  return (
    <div
      className={`item ${props.item.completed ? " completed" : ""}`}
      onClick={() => props.toggleItem(props.item.id)}
    >
      <p style={{ fontFamily: "Indie Flower", fontSize: "2rem" }}>
        {props.item.name}
      </p>
    </div>
  );
}
