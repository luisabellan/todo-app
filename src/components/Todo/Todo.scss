.task {
  display: flex;
  font-family: "Indie Flower";
  input {
    margin-right: 1rem;
  }
  span {
    font-size: "3rem";
  }
}
