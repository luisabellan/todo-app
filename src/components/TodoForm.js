import React, { useState } from "react";

export default function TodoForm(props) {
  const initialState = {
    todoItem1: "",
  };
  const [state, setState] = useState(initialState);

  const handleChanges = (e) => {
    // update state with each keystroke
    // console.log(e)
    setState({ [e.target.name]: e.target.value });
    // console.log(state);
  };

  // class property to submit form
  const submitItem = (e) => {
    e.preventDefault();
    //console.log(props);
    props.addItem(e, state.todoItem1);
    setState({ todoItem1: "" });
  };

  // console.log('rendering form', state.todoItem);

  return (
    <>
      <form onSubmit={submitItem}>
        <h2>New Task</h2>
        <input
          type="text"
          data-testid="todo-input"
          value={state.todoItem1}
          name="todoItem1"
          onChange={handleChanges}
          placeholder="New task"
        />
        /
        <button
          className="add-to-do-btn"
          style={{
            fontFamily: "Jomhuria,sans-serif",
            outline: "none",
            fontSize: "3.2rem",
          }}
        >
          <span>Add</span>
        </button>
        <button
          className="clear-btn"
          onClick={props.clearCompleted}
          style={{
            fontFamily: "Jomhuria,sans-serif",
            outline: "none",
            fontSize: "3.2rem",
          }}
        >
          <span>Clean</span>
        </button>
      </form>
    </>
  );
}
