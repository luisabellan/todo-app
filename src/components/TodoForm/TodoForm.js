import React, { useState } from 'react';
import PropTypes from 'prop-types';

TodoForm.propTypes = {
  addItem: PropTypes.func,
  clearCompleted: PropTypes.func,
  clearAll: PropTypes.func,
};

export default function TodoForm(props) {
  const initialState = {
    todoItem1: ''
  };
  const [state, setState] = useState(initialState);

  const handleChanges = (e) => {
    // update state with each keystroke
    // console.log(e)
    setState({ [e.target.name]: e.target.value });
    // console.log(state);
  };

  // class property to submit form
  const addTodo = (e) => {
    e.preventDefault();
    //console.log(props);
    props.addItem(e, state.todoItem1);
    setState({ todoItem1: '' });
  };

  // console.log('rendering form', state.todoItem);

  return (
    <div>
      <form onSubmit={addTodo} data-testid="form">
        <input
          type="text"
          className="input"
          data-testid="todo-input"
          value={state.todoItem1}
          placeholder="New Task"
          name="todoItem1"
          onChange={handleChanges}
        ></input>
        <div id="buttons">

          <div id="button-left">
            <button className="add-todo-btn" data-testid="add-todo-button">
              <span>Add</span>
            </button>

          </div>

          <div id="button-right">

            <button className="clear-btn" onClick={props.clearCompleted}>
              <span>Completed</span>
            </button>


            <button className="del-btn" onClick={props.clearAll}>
              <span>Clear</span>
            </button>
          </div>

        </div>
      </form>
    </div>
  );
}
