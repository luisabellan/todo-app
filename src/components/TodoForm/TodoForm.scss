h2 {
  padding-top: 3rem;
}
