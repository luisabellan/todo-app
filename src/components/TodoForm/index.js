import React, { useState } from 'react';
import PropTypes from 'prop-types';

TodoForm.propTypes = {
  addItem: PropTypes.func,
  clearCompleted: PropTypes.func
};

export default function TodoForm(props) {
  const initialState = {
    todoItem1: ''
  };
  const [state, setState] = useState(initialState);

  const handleChanges = (e) => {
    // update state with each keystroke
    // console.log(e)
    setState({ [e.target.name]: e.target.value });
    // console.log(state);
  };

  // class property to submit form
  const submitItem = (e) => {
    e.preventDefault();
    //console.log(props);
    props.addItem(e, state.todoItem1);
    setState({ todoItem1: '' });
  };

  // console.log('rendering form', state.todoItem);

  return (
    <>
      <form onSubmit={submitItem}>
        <input
          type="text"
          data-testid="todo-input"
          value={state.todoItem1}
          placeholder="New Task"
          name="todoItem1"
          onChange={handleChanges}
        ></input>

        <button className="add-to-do-btn">
          <span>Add</span>
        </button>
        <button className="clear-btn" onClick={props.clearCompleted}>
          <span>Clean</span>
        </button>
      </form>
    </>
  );
}
