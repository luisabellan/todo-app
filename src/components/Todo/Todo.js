/* eslint-disable no-labels */
/* eslint-disable no-unused-expressions */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './Todo.scss';

Todo.propTypes = {
  todo: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string.isRequired,
    completed: PropTypes.bool,
  }),

  toggleItem: PropTypes.func,
  handleChanges: PropTypes.func
};

export default function Todo(props) {

  const todoInitialState = {
    id: "",
    name: "",
    completed: false
  }

  const [todo, setTodo] = useState(todoInitialState)


  return (
    <div className="checkbox" data-testid="todo">
      <input
        type="checkbox"
        data-testid={`checkbox ${props.todo.name}`}
        name={`item ${props.todo.name}`}
        value={props.todo.name}
        onClick={() => props.toggleItem(props.todo.id)}
        onChange={props.handleChanges}
        defaultChecked={props.todo.completed}
      ></input>
      <span data-testid="todo-output">{props.todo.name}</span>
    </div >
  );
}
