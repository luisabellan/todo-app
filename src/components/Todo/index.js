/* eslint-disable no-labels */
/* eslint-disable no-unused-expressions */
import React from "react";
import PropTypes from "prop-types";
import "./Todo.scss";

/* item: PropTypes.exact({
  id: PropTypes.number.isRequired,
  name: PropTypes.string,
  completed: PropTypes.bool,
}); */

Todo.propTypes = {
  item: PropTypes.object,
  toggleItem: PropTypes.func,
  handleChanges: PropTypes.func,
};

export default function Todo(props) {
  return (
    <div
      /* className={`item ${props.item.completed ? " completed" : ""}`}  */
      className="task"
      /*onClick={() => props.toggleItem(props.item.id)}*/
    >
      <input
        type="checkbox"
        name={`item ${props.item.name}`}
        value={props.item.name}
        onClick={() => props.toggleItem(props.item.id)}
        onChange={props.handleChanges}
        defaultChecked={props.item.completed}
      ></input>
      <span>{props.item.name}</span>
    </div>
  );
}
