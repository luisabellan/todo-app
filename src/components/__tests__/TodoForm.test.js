import React from 'react';
import {
  render,
  screen,
  fireEvent,
  cleanup
  /* getRoles*/
} from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import TodoForm from '../TodoForm';
import Todo from '../Todo';
import App from '../../App';


it('matches snapshot', () => {
  const { asFragment } = render(<TodoForm />);
  expect(asFragment()).toMatchSnapshot();
});

it('renders TodoForm', async () => {
  render(<TodoForm />);
  const todoForm = await screen.findByPlaceholderText(/New task/i);
  const addTodoButton = screen.getByText(/add/i);
  const clearListButton = screen.getByText(/clean/i);

  expect(todoForm).toBeInTheDocument();
  expect(addTodoButton).toBeInTheDocument();
  expect(clearListButton).toBeInTheDocument();
});

it('handle changes of input text', () => {
  let item = "";
  const { getByTestId } = render(<TodoForm />);
  const input = getByTestId('todo-input');
  input.value = item;
  fireEvent.change(input);
  expect(input.value).toBe(item);

  userEvent.type(screen.getByTestId('todo-input'), 'read book')
  expect(screen.getByTestId('todo-input')).toHaveValue('read book')
});

it('add todo', () => {
  let item = {
    todoItem1: ""
  };

  const { getByTestId } = render(<TodoForm />);
  const input = getByTestId('todo-input');
  input.value = item.todoItem1;
  userEvent.type(screen.getByTestId('todo-input'), 'read book')
  expect(screen.getByTestId('todo-input')).toHaveValue('read book')

  userEvent.click(screen.getByText("Add"));
  expect(input.value).toBe(item.todoItem1);

});
