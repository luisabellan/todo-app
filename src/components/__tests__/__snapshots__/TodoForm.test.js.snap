// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="form"
  >
    <form>
      <input
        data-testid="todo-input"
        name="todoItem1"
        placeholder="New Task"
        type="text"
        value=""
      />
      <button
        class="add-to-do-btn"
        data-testid="add-todo-button"
      >
        <span>
          Add
        </span>
      </button>
      <button
        class="clear-btn"
      >
        <span>
          Clean
        </span>
      </button>
    </form>
  </div>
</DocumentFragment>
`;
